packages/viewer/src/lib/types/frame.ts

```diff
@@ -171,27 +201,38 @@ export class FramePerspectiveCamera
     return this.cameraMatrices;
   }
 
-  public get direction(): Vector3.Vector3 {
-    return Vector3.normalize(Vector3.subtract(this.lookAt, this.position));
-  }
-
-  public get worldMatrix(): Matrix4.Matrix4 {
-    return this.computeCameraMatrices().worldMatrix;
-  }
-
-  public get viewMatrix(): Matrix4.Matrix4 {
-    return this.computeCameraMatrices().viewMatrix;
-  }
-
-  public get projectionMatrix(): Matrix4.Matrix4 {
-    return this.computeCameraMatrices().projectionMatrix;
+  /**
+   * Checks if the given point, in world space, is behind the near plane
+   * of the camera.
+   *
+   * @param world A point in world space.
+   * @returns `true` if the point is behind the camera.
+   */
+  public isPointBehindNear(world: Vector3.Vector3): boolean {
+    const { position, direction, near } = this;
+    const dist = Vector3.dot(Vector3.subtract(world, position), direction);
+    return dist < near;
   }
 
-  public get projectionMatrixInverse(): Matrix4.Matrix4 {
-    return this.computeCameraMatrices().projectionMatrixInverse;
-  }
-
-  public get projectionViewMatrix(): Matrix4.Matrix4 {
-    return this.computeCameraMatrices().projectionViewMatrix;
+  /**
+   * Returns a point on the near plane that intersects with `line`. If `line`
+   * does not intersect, then `undefined` is returned.
+   *
+   * @param line The line to intersect.
+   * @returns A point in world space, or `undefined` if the line does not
+   * intersect with the near plane.
+   */
+  public intersectLineWithNear(line: Line3.Line3): Vector3.Vector3 | undefined {
+    const { position, lookAt, direction, near } = this;
+
+    const vs = Vector3.subtract(line.start, lookAt);
+    const ve = Vector3.subtract(line.end, lookAt);
```

Comment:
Shouldn't these be subtracting `position`?

Reply:
Resolving to consolidate conversation in other thread.